lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -314,7 +321,18 @@ defmodule ExUnit.Case do
       Module.put_attribute(mod, :ex_unit_test_names, Map.put(test_names, name, true))
     end
 
-    Module.delete_attribute(mod, :tag)
+    Enum.each [:tag | test_attributes], fn(attribute) ->
+      Module.delete_attribute(mod, attribute)
+    end
+  end
+
+  @doc """
+  """
+  defmacro register_attribute(mod, new, opts \\ []) when is_atom(new) do
```

Comment:
register_attribute should be a function and receive the current module. You
can call put_attribute to update the module attribute. Other than that, I
am really liking the direction this is going, we will definitely get this
in.



-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*